stan/math/rev/fun/log_determinant_spd.hpp

```diff
@@ -16,91 +16,43 @@ namespace math {
 /**
  * Returns the log det of a symmetric, positive-definite matrix
  *
- * @tparam EigMat Type of the matrix
+ * @tparam T Type is an Eigen Matrix
  * @param m a symmetric, positive-definite matrix
  * @return The log determinant of the specified matrix
  */
-template <typename EigMat, require_eigen_vt<is_var, EigMat>* = nullptr>
-inline var log_determinant_spd(const EigMat& m) {
+template <typename T, require_rev_matrix_t<T>* = nullptr>
```

Comment:
(optional) I prefer slightly more informative names, like `Mat` or even just `M` for matrix